substrate/utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -761,13 +762,35 @@ fn list_benchmark(
 		Vec<(BenchmarkParameter, u32, u32)>,
 		Vec<(String, String)>,
 	)>,
+	pallets_only: bool,
 ) {
 	// Sort and de-dub by pallet and function name.
-	benchmarks_to_run.sort_by(|(pa, sa, _, _), (pb, sb, _, _)| (pa, sa).cmp(&(pb, sb)));
-	benchmarks_to_run.dedup_by(|(pa, sa, _, _), (pb, sb, _, _)| (pa, sa) == (pb, sb));
-
-	println!("pallet, benchmark");
-	for (pallet, extrinsic, _, _) in benchmarks_to_run {
-		println!("{}, {}", String::from_utf8_lossy(&pallet), String::from_utf8_lossy(&extrinsic));
+	benchmarks_to_run.sort_by(
+		|(pa, sa, _, _), (pb, sb, _, _)| {
+			if pallets_only {
+				pa.cmp(pb)
+			} else {
+				(pa, sa).cmp(&(pb, sb))
+			}
+		},
+	);
+	benchmarks_to_run.dedup_by(
+		|(pa, sa, _, _), (pb, sb, _, _)| if pallets_only { pa == pb } else { (pa, sa) == (pb, sb) },
+	);
```

Comment:
I think we can improve the code here by using `BTreeMap<String, BTreeSet<String>>`.  
It is automatically sorted and de-duplicated (and should make the code more readable).